### PR TITLE
Sbhivandkar 424897 bug self service registration

### PR DESCRIPTION
**Pull request by shreyasbhivandkar-okta**
## Description:
Save state enabled on Self Service Registration page during model.save stage.

## PR Checklist

- [x] Have you verified the basic functionality for this change?

- [x] Added unit tests?

- [ ] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Screenshot/Video:
Bug:
SelfServiceRegistration

Fix:

https://user-images.githubusercontent.com/88052070/138795902-ed90d81e-c06b-4f88-9dbd-03829ff13da9.mov

Bacon - https://bacon-go.aue1e.saasure.net/commits?artifact=okta-signin-widget&branch=sbhivandkar-424897-bug-self-service-registration

### Reviewers:
@conorhanrahan-okta 
cc: @haishengwu-okta @nikhilvenkatraman-okta @lesterchoi-okta 

### Issue:

- [OKTA-424897](https://oktainc.atlassian.net/browse/OKTA-424897)

### CODE REVIEW

File: test/unit/spec/Registration_spec.js [hunk 1]
```diff
@@ -994,5 +996,45 @@ Expect.describe('Registration', function() {
...
+          test.form.setLastname('lastName');
+          test.form.setReferrer('referrer');
+          expect($('input.button-primary').length).toBe(1);
+          expect($('input.button-primary.btn-disabled').length).toBe(0);
+          test.form.submit();
+          return new Promise(resolve => setTimeout(resolve, 100));
+        }).then(() => {
+          expect($('input.button-primary.btn-disabled').length).toBe(1);
+        });
+    });
```
Comment (shreyasbhivandkar-okta) [inline test/unit/spec/Registration_spec.js:1038]: @conorhanrahan-okta The test has been added. Kindly review, thanks.

Comment (conorhanrahan-okta) [reply test/unit/spec/Registration_spec.js:1038]: Looks good to me!

**Review by conorhanrahan-okta — APPROVED** [review]
Thanks for addressing all my comments, nice work on this fix!

File: playground/mocks/data/idp/idx/enroll-profile-submit.json [hunk 2]
```diff
@@ -1,84 +1,84 @@
...
-      "id":"0oay6fLeui13rEyTj0g3"
+  "app": {
+    "type": "object",
+    "value": {
+      "name": "okta_enduser",
+      "label": "Okta Dashboard",
+      "id": "0oay6fLeui13rEyTj0g3"
     }
   }
 }
```
Comment (shreyasbhivandkar-okta) [inline playground/mocks/data/idp/idx/enroll-profile-submit.json:84]: @fangshi-okta this is the file I was talking about. I just formatted the file without changing the content, thanks.

Comment (fangshi-okta) [reply playground/mocks/data/idp/idx/enroll-profile-submit.json:84]: Thank you!

**Pull request closed without merging**